### PR TITLE
don't rotate a rigidbody around locked axis by AddMovement

### PR DESCRIPTION
**Pull request by nothingTVatYT**
This fix will prevent rotating a rigidbody around locked axis by the root motion delta.

### CODE REVIEW

File: Source/Engine/Physics/Actors/RigidBody.cpp [hunk 1]
```diff
@@ -300,6 +300,46 @@ void RigidBody::ClosestPoint(const Vector3& position, Vector3& result) const
     }
 }
 
+void RigidBody::AddMovement(const Vector3& translation, const Quaternion& rotation)
+{
+    // filter rotation according to constraints
+    Quaternion allowedRotation;
+    if (static_cast<unsigned int>(GetConstraints()) & static_cast<unsigned int>(RigidbodyConstraints::LockRotation))
```
Comment (mafiesto4) [inline Source/Engine/Physics/Actors/RigidBody.cpp:307]: Use `EnumHasAllFlags(GetConstraints(). RigidbodyConstraints::LockRotation)` for readability (the same for translation).

File: Source/Engine/Physics/Actors/RigidBody.cpp [hunk 2]
```diff
@@ -300,6 +300,46 @@ void RigidBody::ClosestPoint(const Vector3& position, Vector3& result) const
...
+        allowedRotation = Quaternion::Identity;
+    else
+    {
+        Float3 euler = rotation.GetEuler();
+        const auto constraints = static_cast<unsigned int>(GetConstraints());
+        if (constraints & static_cast<unsigned int>(RigidbodyConstraints::LockRotationX))
```
Comment (mafiesto4) [inline Source/Engine/Physics/Actors/RigidBody.cpp:313]: Use `EnumHasAnyFlags(GetConstraints(). RigidbodyConstraints::LockRotationX)` for readability (the same for Y/Z axes and translation axes too).

File: Source/Engine/Physics/Actors/RigidBody.h [hunk 3]
```diff
@@ -430,6 +430,12 @@ class FLAXENGINE_API RigidBody : public Actor, public IPhysicsActor
     /// <param name="result">The result point on the rigidbody shape that is closest to the specified location.</param>
     API_FUNCTION() void ClosestPoint(const Vector3& position, API_PARAM(Out) Vector3& result) const;
 
+    /// <summary>
+    /// Moves and rotates the rigidbody in world space within the limits of defined constraints.
+    /// </summary>
+    /// <param name="translation">The translation vector.</param>
+    /// <param name="rotation">The rotation quaternion.</param>
+    API_FUNCTION() void AddMovement(const Vector3& translation, const Quaternion& rotation) override;
```
Comment (mafiesto4) [inline Source/Engine/Physics/Actors/RigidBody.h:438]: Move it down below `// [Actor]` and remove not needed XML doc comments and `API_FUNCTION()` so it's just pure override (base Actor class has proper docs and exposed scripting info).

File: Source/Engine/Level/Actors/AnimatedModel.cpp [hunk 4]
```diff
@@ -15,6 +15,7 @@
 #include "Engine/Graphics/Models/MeshDeformation.h"
 #include "Engine/Level/Scene/Scene.h"
 #include "Engine/Level/SceneObjectsFactory.h"
+#include "Engine/Physics/Actors/RigidBody.h"
```
Comment (mafiesto4) [inline Source/Engine/Level/Actors/AnimatedModel.cpp:18]: Remove unused include

**Review by mafiesto4 — CHANGES_REQUESTED** [review]

**Comment by nothingTVatYT** [comment]
All requested changes done - assuming I understood you correctly that is.

**Review by mafiesto4 — APPROVED** [review]

**Pull request merged**